src/Nancy/Diagnostics/DiagnosticsHook.cs

```diff
@@ -188,7 +188,7 @@ private static Response ExecuteDiagnostics(NancyContext ctx, IRouteResolver rout
 
             if (resolveResult.After != null)
             {
-                resolveResult.After.Invoke(ctx, CancellationToken);
+                resolveResult.After.Invoke(ctx, CancellationToken).Wait();

```

Comment:
Yeah, we must not forget that we have a pretty half-assed async story right now. I think we really need to look at getting everything async soon -- before we release 2.0.